File: test-suite/src/test/java/io/micronaut/docs/http/bind/binders/MyBindingAnnotation.java
package io.micronaut.docs.http.bind.binders;

// tag::class[]
import io.micronaut.context.annotation.AliasFor;
import io.micronaut.core.bind.annotation.Bindable;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.ANNOTATION_TYPE})
@Retention(RUNTIME)
@Bindable //<1>
public @interface MyBindingAnnotation {
    @AliasFor(annotation = Bindable.class, member = "value")
    String value() default "";
}

// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/bind/binders/MyBindingAnnotation.kt
package io.micronaut.docs.http.bind.binders

// tag::class[]
import io.micronaut.core.bind.annotation.Bindable

@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Bindable //<1>
annotation class MyBindingAnnotation
// end::class[]